test/jdk/java/awt/ScrollPane/bug8077409Test.java

```diff
@@ -77,41 +88,52 @@ protected void processKeyEvent(KeyEvent e) {
+    try {
+      obj.setLocationRelativeTo(null);
+      Toolkit.getDefaultToolkit().addAWTEventListener(e -> {
         KeyEvent keyEvent = (KeyEvent) e;
-        if(keyEvent.getID() == KeyEvent.KEY_RELEASED) {
-            if (keyEvent.getKeyCode() == KeyEvent.VK_1) {
-              System.out.println(obj.pane.toString());
-              System.out.println("obj.myCanvas.pos: " + obj.myCanvas.getBounds());
-              System.out.println(obj.myCanvas.toString());
-            }  else if (keyEvent.getKeyCode() == KeyEvent.VK_2) {
-              obj.repaint();
-           } else if (keyEvent.getKeyCode() == KeyEvent.VK_DOWN) {
-              Point scrollPosition = obj.pane.getScrollPosition();
-              scrollPosition.translate(0, 1);
-              obj.pane.setScrollPosition(scrollPosition);
-            } else if (keyEvent.getKeyCode() == KeyEvent.VK_UP) {
-              Point scrollPosition = obj.pane.getScrollPosition();
-              scrollPosition.translate(0, -1);
-              obj.pane.setScrollPosition(scrollPosition);
-            } else if (keyEvent.getKeyCode() == KeyEvent.VK_SPACE) {
-              obj.pane.validate();
-            }
+        if (keyEvent.getID() == KeyEvent.KEY_RELEASED) {
+          if (keyEvent.getKeyCode() == KeyEvent.VK_1) {
+            System.out.println(obj.pane.toString());
+            System.out.println("obj.myCanvas.pos: " + obj.myCanvas.getBounds());
+            System.out.println(obj.myCanvas.toString());
+          } else if (keyEvent.getKeyCode() == KeyEvent.VK_2) {
+            obj.repaint();
+          } else if (keyEvent.getKeyCode() == KeyEvent.VK_DOWN) {
+            Point scrollPosition = obj.pane.getScrollPosition();
+            scrollPosition.translate(0, 1);
+            obj.pane.setScrollPosition(scrollPosition);
+          } else if (keyEvent.getKeyCode() == KeyEvent.VK_UP) {
+            Point scrollPosition = obj.pane.getScrollPosition();
+            scrollPosition.translate(0, -1);
+            obj.pane.setScrollPosition(scrollPosition);
+          } else if (keyEvent.getKeyCode() == KeyEvent.VK_SPACE) {
+            obj.pane.validate();
           }
         }
-    }, AWTEvent.KEY_EVENT_MASK);
+      }, AWTEvent.KEY_EVENT_MASK);
+      obj.setVisible(true);
+      Robot robot = new Robot();
+      robot.waitForIdle();
+      robot.delay(300);
       Point scrollPosition = obj.pane.getScrollPosition();
       scrollPosition.translate(0, 1);
       obj.pane.setScrollPosition(scrollPosition);
 
       int y = obj.pane.getComponent(0).getLocation().y;
+      robot.waitForIdle();
+      robot.delay(300);
       obj.pane.validate();
-      if(y != obj.pane.getComponent(0).getLocation().y){
-          throw new RuntimeException("Wrong position of component in ScrollPane");
+      robot.waitForIdle();
+      robot.delay(300);
+      if (y != obj.pane.getComponent(0).getLocation().y) {
+        throw new RuntimeException("Wrong position of component in ScrollPane");
```

Comment:
Looks like the code uses a non-standard two-space indentation.

Reply:
Changed the code indentation